hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -609,7 +663,10 @@ func deleteMachineHealthCheck(ctx context.Context, c client.Client, mhc *capiv1.
 }
 
 func (r *NodePoolReconciler) delete(ctx context.Context, nodePool *hyperv1.NodePool, infraID, controlPlaneNamespace string) error {
+	delete(r.hostedClusterCachesTracker.caches, client.ObjectKeyFromObject(nodePool))
```

Comment:
I think we're only using one reconciler thread, but should this map write be guarded with a mutex anyway just to be safe?

Reply:
done!